gluten-ut/spark40/src/test/scala/org/apache/spark/sql/GlutenDataFrameFunctionsSuite.scala

```diff
@@ -49,4 +52,230 @@ class GlutenDataFrameFunctionsSuite extends DataFrameFunctionsSuite with GlutenS
       false
     )
   }
+
+  testGluten("map_concat function") {
+    val df1 = Seq(
+      (Map[Int, Int](1 -> 100, 2 -> 200), Map[Int, Int](3 -> 300, 4 -> 400)),
+      (Map[Int, Int](1 -> 100, 2 -> 200), Map[Int, Int](3 -> 300, 1 -> 400)),
+      (null, Map[Int, Int](3 -> 300, 4 -> 400))
+    ).toDF("map1", "map2")
+
+    val expected1a = Seq(
+      Row(Map(1 -> 100, 2 -> 200, 3 -> 300, 4 -> 400)),
+      Row(Map(1 -> 400, 2 -> 200, 3 -> 300)),
+      Row(null)
+    )
+
+    intercept[SparkException](df1.selectExpr("map_concat(map1, map2)").collect())
```

Comment:
I think there are similar tests found when i work on the first patch to enable unit tests. Is it too heavy to modify the `map_concat `function to throw the right exception?

Reply:
I haven't found anything similar yet. Would the impact be greater if GlutenException were changed to SparkRuntimeException?

Reply:
Can GlutenException inherit SparkRuntimeException? Now it inherits RuntimeException, I'm not sure if it is more feasible.